<launch>

	<!-- need to change your bag files to be named "main.bag", "front.bag", "rear.bag" -->

	<!-- choose sensor: "main", "front", "rear" -->
	<arg name="sensor" default="main" />

	<param name="/use_sim_time" value="true"/>

	<node pkg="rosbag" type="play" name="$(eval arg('sensor') + '_lidar' )" output="screen" args="$(eval '--clock /home/phil/subt_data/' + arg('sensor') + '.bag -r 2')"/>

	<node pkg="octomap_server" type="octomap_server_node" name="octomap_server">
		<param name="resolution" value="0.1" />
		
		<!-- fixed map frame (set to 'map' if SLAM or localization running!) -->
		<param name="frame_id" type="string" value="husky4/map" />
		
		<!-- maximum range to integrate (speedup!) -->
		<param name="sensor_model/max_range" value="15.0" />

		<param name="height_map" type="bool" value="false" />
		<param name="color/r" value="1.0" />
		<param name="color/g" value="0.0" />
		<param name="color/b" value="0.0" />
		<param name="color/a" value="0.3" />
		
		<!-- data source to integrate (PointCloud2) -->
		<remap from="cloud_in" to="/husky4/velodyne_points" if="$(eval arg('sensor') == 'main')" />
		<remap from="cloud_in" to="/husky4/velodyne_front/velodyne_points" if="$(eval arg('sensor') == 'front')" />
		<remap from="cloud_in" to="/husky4/velodyne_rear/velodyne_points" if="$(eval arg('sensor') == 'rear')" />

	</node>

	<node pkg="visuals" type="tf_transform" name="$(eval arg('sensor') + '_tf_tranformer' )" >
		<param name = "sensor" value = "$(eval arg('sensor'))" />

	</node>

	<node pkg="rviz" type="rviz" name="rviz" />

</launch>